service/matching/matcher.go

```diff
@@ -266,28 +280,32 @@ forLoop:
 // Poll blocks until a task is found or context deadline is exceeded
 // On success, the returned task could be a query task or a regular task
 // Returns ErrNoTasks when context deadline is exceeded
-func (tm *TaskMatcher) Poll(ctx context.Context) (*InternalTask, error) {
+func (tm *TaskMatcher) Poll(ctx context.Context, isolationGroup string) (*InternalTask, error) {
+	isolatedTaskC, ok := tm.isolatedTaskC[isolationGroup]
+	if !ok && isolationGroup != "" {
+		return nil, fmt.Errorf("invalid isolation group: %s", isolationGroup)
```

Comment:
Nit: this probably should be a `types.BadRequestError{}` or some kind of error to indicate that it's a caller problem, not a server-side problem, otherwise it'll burn our SLAs

Reply:
good point.